addon/components/inputs/datetime.js

```diff
@@ -13,12 +14,40 @@ export default AbstractInput.extend({
 
   layout,
 
+  // == State Properties =======================================================
+
+  getDefaultProps () {
+    if (this.get('value')) {
+      var arr = this.get('value').split('T')
+
+      return {
+        date: arr[0] || '',
+        time: arr[1] || ''
+      }
+    }
+
+    return {
+      date: '',
+      time: ''
+    }
+  },
+
   // == Computed Properties ====================================================
 
   // == Functions ==============================================================
 
+  parseValue (value) {
+    if (value.includes('-')) {
+      this.set('date', value)
+    } else {
+      this.set('time', 'T' + value)
+    }
+    return this.get('date') + this.get('time')
+  },
```

Comment:
I've tried implementing your recommendation of computed properties but the functionality isn't the same. For example updating the time only is not possible, you would need to update the date and time. 
This is a problem for example if a user wants to search on time through Jan 1, 2016, they would need to put the date first then time, and then if they wish to update the time, they would need to reselect the date.. 
The code I placed before had both functionality, i can't seem to figure out how to make this work with computed..